File: БР1.1/Зотеев Максим/labs/lab2/services/rental-service/src/entities/Rental.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn } from "typeorm";

export type RentalStatus = "active" | "completed" | "cancelled";

@Entity("rentals")
export class Rental {
  @PrimaryGeneratedColumn({ type: "bigint" })
  id: string;

  // Внешние идентификаторы — указывают на сущности из других сервисов.
  // FK не задаём: пересечение БД сервисов запрещено по архитектуре.
  @Column({ name: "property_id", type: "bigint" })
  propertyId: string;

  @Column({ name: "tenant_id", type: "bigint" })
  tenantId: string;

  @Column({ name: "owner_id", type: "bigint" })
  ownerId: string;

  // Снапшоты на момент создания сделки — чтобы не дёргать property-service
  // ради базовой информации в списке "мои сделки".
  @Column({ name: "property_title_snapshot" })
  propertyTitleSnapshot: string;

  @Column({ name: "property_city_snapshot" })
  propertyCitySnapshot: string;

  @Column({ name: "price_per_month_snapshot", type: "numeric", precision: 12, scale: 2 })
  pricePerMonthSnapshot: string;

  @Column({ name: "start_date", type: "date" })
  startDate: string;

  @Column({ name: "end_date", type: "date", nullable: true })
  endDate: string;

  @Column({ type: "varchar", length: 16, default: "active" })
  status: RentalStatus;

  @CreateDateColumn({ name: "created_at" })
  createdAt: Date;
}

File: БР1.1/Данилова Анастасия/labs/rental-platform/services/rental-service/internal/models/rental.go
package models

import "time"

type RentalStatus string

const (
	StatusPending   RentalStatus = "PENDING"
	StatusApproved  RentalStatus = "APPROVED"
	StatusRejected  RentalStatus = "REJECTED"
	StatusActive    RentalStatus = "ACTIVE"
	StatusCompleted RentalStatus = "COMPLETED"
	StatusCancelled RentalStatus = "CANCELLED"
)

type Rental struct {
	ID         uint         `gorm:"primaryKey" json:"id"`
	PropertyID uint         `gorm:"not null;index" json:"property_id"`
	TenantID   uint         `gorm:"not null;index" json:"tenant_id"`
	LandlordID uint         `gorm:"not null;index" json:"landlord_id"`
	StartDate  time.Time    `gorm:"type:date;not null" json:"start_date"`
	EndDate    time.Time    `gorm:"type:date;not null" json:"end_date"`
	TotalPrice int          `gorm:"not null" json:"total_price"`
	Status     RentalStatus `gorm:"type:varchar(20);not null;index" json:"status"`
	CreatedAt  time.Time    `json:"created_at"`
	UpdatedAt  time.Time    `json:"updated_at"`
}

func (Rental) TableName() string {
	return "rentals"
}
